calyx/src/passes/merge_static_par.rs

```diff
@@ -0,0 +1,120 @@
+use crate::ir::{
+    self,
+    traversal::{Action, Named, VisResult, Visitor}, RRC,
+};
+use std::{collections::HashMap};
+use itertools::partition;
+use crate::ir::Enable;
+use std::rc::Rc;
+use crate::ir::Attributes;
+
+
+/// Pass to do something
+#[derive(Default)]
+pub struct MergeStaticPar ;
+
+impl Named for MergeStaticPar {
+    fn name() -> &'static str {
+        "merge-static-par"
+    }
+
+    fn description() -> &'static str {
+        "merge static pars when they have the same static time"
+    }
+}
+
+impl Visitor for MergeStaticPar {
+    fn finish_par(
+        &mut self,
+        s: &mut ir::Par,
+        _comp: &mut ir::Component,
+        _sigs: &ir::LibrarySignatures,
+        _comps: &[ir::Component],
+    ) -> VisResult {
+        let mut static_group:HashMap<u64, Vec<RRC<ir::Group>>> = HashMap::new();
+        let idx = partition(&mut s.stmts, 
+            |stmt| matches!(stmt, ir::Control::Enable(_)));
+
+        let e_stmts: Vec<_> = s.stmts.drain(0..idx).collect();
```

Comment:
Use the partition method defined by itertools instead of using the standard libraries mutating partition method: https://docs.rs/itertools/latest/itertools/fn.partition.html

Reply:
this is the itertools partition

Reply:
Ah, it should be iterator partition then: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition